content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,519 @@
+
+*I will be using the popular iris data set for the tutorial, you can get it almost anywhere but I downloaded this from [Kaggle](https://www.kaggle.com/vikrishnan/iris-dataset).*
+
+#### Load the data into the notebook
+```python
+df = pd.read_csv("/content/drive/MyDrive/IRIS.csv")
+```
+```python
+df.head()
+```
+![df.head.png](df.head.png)
+On checking the first five, it seems all the species of flower are setosa, but the iris dataset has 2 more samples apart from that, to be sure, let's run df.sample() to visualize samples randomly.
+```python
+df.sample(10)
+```
+![sampleten](df.sample.png)
+
+Always check for missing values to avoid future complications.
+```python
+df.isnull().sum()
+```
+![isnull](df.isnull.png)
+
+We can be sure now that there are 3 classes of species and there are no missing values. 
+Let's proceed with the manipulation.
+
+Let's use petal width to compare species for our Chi-Square Test. 
+
+The first step is to check the summary of petal_width 
+
+Secondly, use the information derived to convert it to a categorical variable.
+```python
+df.petal_width.describe()
+```
+![petalwidthdescribe](petalwidthdescribe.png)
+
+Since, the operation we are going to be performing are mathematical operations and they don't work with letters or words except numbers. The Chi-square test helps in determining if there is a notable difference between the observed and normal frequencies in one or more categories, the values of petal_width must be changed into categories of zeros and ones using the 50% percentile which is the second quartile as the yardstick. There are clear [differences](https://byjus.com/maths/difference-between-percentage-and-percentile/) between percentage and percentile. As percentage is just a fraction of a hundred, the percentile is more concerned with ranks, it can be termed as the percentage of values that fall below a certain value mark. It is rather used to find the position of a value in a given set of values. For example, if a girl has the 50th percentile on an examination of 200, we can say that by scoring 200 marks, she has higher marks than 50% of the remaining class.
+
+#### How to calculate percentile.
+There are two conditions that must be satisfied.
+1. Data must be arranged in ascending order.
+2. if the index is not an integer, it must be rounded up.
+```
+np.percentile(df.petal_width, 50)
+```
+![percentile](percentile.png)
+
+ A quartile is another term-related percentile, it can be called a specified percentile, we have 1st quartile as 25% percentile, second quartile as 50%, and 3rd as 75% percentile. You can as well read this [article](https://datascienceupskill.com/statistics/percentiles-and-quartiles) to get a better grip.  
+
+```python
+def petal_cat(df):
+ if df['petal_width'] <= 1.3:
+ return 0
+ elif df['petal_width'] > 1.3:
+ return 1
+ else:
+ return 'indifferent'
+
+df['petal_width_new'] = df.apply(petal_cat, axis=1)  
+```
```

Comment:
Do indent your code properly.
The `if` block and the `return` block aren't indented properly.

Reply:
done